docs/source/how_to/how_to_algorithm_selection.ipynb

```diff
@@ -5,26 +5,54 @@
    "metadata": {},
    "source": [
     "(how-to-select-algorithms)=\n",
+    "# How to select a local optimizer\n",
     "\n",
-    "# Which optimizer to use\n",
+    "This guide explains how to choose a local optimizer that works well for your problem. \n",
+    "Depending on your [strategy for global optimization](how_to_globalization.ipynb) it \n",
+    "is also relevant for global optimization problems. \n",
     "\n",
-    "This is a very very very short and oversimplifying guide on selecting an optimization algorithm based on a minimum of information. \n",
+    "## Important facts \n",
     "\n",
+    "- There is no optimizer that works well for all problems \n",
+    "- Making the right choice can lead to enormous speedups\n",
+    "- Making the wrong choice can mean that you don't solve your problem at all; Sometimes, \n",
+    "optimizers fail silently!\n",
     "\n",
-    "To select an optimizer, you need to answer two questions:\n",
     "\n",
-    "1. Is your criterion function differentiable?\n",
+    "## The four steps for selecting algorithms\n",
     "\n",
-    "2. Do you have a nonlinear least squares structure (i.e. do you sum some kind of squared residuals at the end of your criterion function)?"
+    "Algorithm selection is a mix of theory and experimentation. We recommend the following \n",
+    "for steps:\n",
+    "\n",
+    "1. **Theory**: Select three to 5 candidate algorithms based on the properties \n",
+    "of your problem. Below we provide a simple decision tree for this step.\n",
+    "2. **Experiments**: Run the candidate algorithms for a small number of function \n",
+    "evaluations. As a rule of thumb, use between `n_params` and `10 * n_params`\n",
+    "evaluations. \n",
+    "3. **Comparison**: Compare the results in a *criterion plot*.\n",
+    "4. **Optimization**: Re-run the algorithm with the best results until \n",
```

Comment:
I agree that 2 and three could be merged. 

I think the optimization should be an explicit step in the guide because we start from the best result from the experiments and I think this should be shown. 

Generally I prefer more but simpler steps. This guide is written for people who are afraid of selecting an algorithm, so I want to make it as mechanical as possible.

Reply:
Switched to three steps now.